### PR TITLE
LPS-201938 Accessibility of the clay not works with some select options

### PR DESCRIPTION
**Pull request by Artartur**
Related Issue: [https://liferay.atlassian.net/browse/LPS-201938](LPS-201938)

### CODE REVIEW

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx [hunk 1]
```diff
@@ -596,45 +596,67 @@ export default function ObjectFieldFormBase({
...
+										listTypeDefinitionsURL,
+										'_blank'
+									);
+								}}
+							>
+								<span className="icon">
+									{Liferay.Language.get('manage-picklists')}
+								</span>
+
+								<ClayIcon symbol="shortcut" />
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx:656]: Hey @Artartur, reloading all of these two buttons is unnecessary as the only component that needs to be reloaded is the SingleSelect, so you can remove it from the condition, you can also remove the div around the entire inside of the condition, the condition itself will be only, please tell me if you have any question about this:

```tsx
{reload ? (
	<ClayLoadingIndicator
		displayType="secondary"
		size="sm"
	/>
) : (
	<SingleSelect
		className="lfr-objects__object-field-form-base-picklist-select-field"
		disabled={disabled}
		error={errors.listTypeDefinitionId}
		id="objectFieldFormBase"
		items={listTypeDefinitionsItems}
		label={Liferay.Language.get('picklist')}
		onSelectionChange={(value) => {
			const selectedListTypeDefinition = listTypeDefinitions.find(
				({externalReferenceCode}) =>
					externalReferenceCode === value
			);
			if (selectedListTypeDefinition) {
				setValues({
					listTypeDefinitionExternalReferenceCode:
						selectedListTypeDefinition.externalReferenceCode,
					listTypeDefinitionId:
						selectedListTypeDefinition.id,
					objectFieldSettings: removeFieldSettings(
						['defaultValue', 'stateFlow'],
						values
					),
				});

				if (onSubmit) {
					onSubmit({
						...values,
						listTypeDefinitionExternalReferenceCode:
							selectedListTypeDefinition.externalReferenceCode,
						listTypeDefinitionId:
							selectedListTypeDefinition.id,
						objectFieldSettings: removeFieldSettings(
							[
								'defaultValue',
								'stateFlow',
							],
							values
						),
					});
				}
			}
		}}
		selectedKey={
			selectedListTypeDefinitionExternalReferenceCode
		}
	/>
)}
```

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx [hunk 2]
```diff
@@ -392,9 +394,14 @@ export default function ObjectFieldFormBase({
 		};
 
 		makeFetch();
+		setReload(true);
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx:397]: @Artartur, As for now we need to update this state only for the Picklist case, you can move this `setReload(true);` for the `getObjectFieldSettingsByBusinessType` function inside the Picklist and MultiselectPicklist condition:

```tsx
...

if (businessType === 'Picklist' || businessType === 'MultiselectPicklist') {
    setReload(true);
    
    updateListTypeDefinitions(setListTypeDefinitions);
}

...
```

**Review by danvb112 — CHANGES_REQUESTED** [review]
Hey @Artartur, Great job on this bug! I have some suggestions!  :smile:

**Comment by Artartur** [comment]
ci:test:sf

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx [hunk 3]
```diff
@@ -201,7 +202,8 @@ export default function ObjectFieldFormBase({
 		oneToManyObjectRelationship,
 		setOneToManyObjectRelationship,
 	] = useState<TObjectRelationship>();
-	const [selectedOutputValue, setSelectedOutputValue] = useState<string>('');
+	const [reload, setReload] = useState(false);
```
[the remark this answers is not in the document]
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx]: Done!

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx [hunk 4]
```diff
@@ -392,9 +394,14 @@ export default function ObjectFieldFormBase({
 		};
 
 		makeFetch();
+		setReload(true);
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [objectRelationshipId, values.businessType]);
 
+	useEffect(() => {
+		setTimeout(() => setReload(false), 200);
+	}, [reload]);
```
[the remark this answers is not in the document]
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx]: Done!

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx (hunk 1 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx]: Done!

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx (hunk 2 above)
Comment (Artartur) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx]: Done!

**Review by danvb112 — APPROVED** [review]
LGTM! :tada:

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by davisaints** [comment]
No regressions found!

**Comment by carolmariaabb** [comment]
ci:forward:force

**Pull request closed without merging**